### ISSUE TITLE
Can't not save record if in record data contain emoji character

### ISSUE THREAD

**Issue by anhnhatuit** [salesagility/SuiteCRM#6447]
#### Issue
I've created new record in Accounts module. In description field. I have type special character is emoji. The system display blank page and I can not found it on Account list then.
**Note:** I tried on my local with Windows 10, MSSQL 2016, SQL server driver I can save it. The problem only faced on Centos.

#### Expected Behavior

Should support create record contain emoji character.
#### Steps to Reproduce

1. Go to Accounts module > Create record
2. Type '😀🤗' character into any text field.
3. Hit Save

#### Your Environment

SuiteCRM Version used: 7.8.8 LTS
Browser name and version (e.g. Chrome Version 51.0.2704.63 (64-bit)):
Environment name and version (e.g. MySQL, PHP 7):
**Operating System and version Centos 7:
DBMS: SQL Server 2016
Driver to connect to DB: FreeTDSManager**

**Comment by DJUserkent** [comment]
This also happens on the Demo at https://suitecrm.com/demo.
In general - in very many pages on SuiteCRM pasting an Emoji icon  causes a Save to fail - no record adding/edited in database.
An error page is served  - white page saying 'Database failure. Please refer to suitecrm.log for details.'

It is due to the emoticon icon character taking more bytes than expected.  (NB - using just text characters to fake an emoticon eg :)   does NOT cause the error -  ONLY the emoticon icon)

Eg - a new Call will not be saved if an emoji is copied into the Description box  - or the Subject box.
An Opp will not be saved if an emoticon is in the Subject or Description

It is probably the  root cause behind this Issue: https://github.com/salesagility/SuiteCRM/issues/2915

**Comment by DJUserkent** [comment]
also proved the exact same problem on an instance of SugarCRM 6.20, on Linux/MySQL

**Comment by sasha2002** [comment]
2 years and no solutions......

If you have any smiles it fails.

`'',0,0,'school 🙂',0): MySQL error 1366: Incorrect string value: '\xF0\x9F\x99\x82' for column 'description' at row 1`

Maybe a fix
https://github.com/salesagility/SuiteCRM/pull/3196/commits

Have these fix set but no luck.
Anyone managed this ?

**Comment by labcerouno** [comment]
As far as I have investigated, you have to convert each column of each table, being careful with the size of the fields (some of them should be changed). That would be an herculean task! An automatic script would be a life saver, if possible. 

Does anyone confirm this is the only way to solve this problem?

**Comment by sasha2002** [comment]
Nope, Column description is utf8mb4 and no luck, still face the issue :(
I will try to convert each column but this will be ........

**Comment by sasha2002** [comment]
Hey, Can someone put URGENT priority to this task. A lot of mails can't be saved because of....

**Comment by sasha2002** [comment]
Found that sugar 9 have fix for our issue, when suite will upgrade ??? 
https://community.sugarcrm.com/community/developer/blog/2019/04/09/sugar-9-has-arrived
"Emoji support on MySQL (utf8mb4 conversion)"
@cameronblaikie  can you say something?

**Comment by pgorod** [comment]
This thread _may_ contain interesting clues for a solution to this

https://community.sugarcrm.com/thread/30853-emails-with-new-smiley-emoji-not-archived-error

but somebody would have to try those tips and see if they're worth pursuing or not...
